src/pkg/cli/update.go

```diff
@@ -0,0 +1,55 @@
+package cli
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/DefangLabs/defang/src/pkg/term"
+)
+
+func Update(ctx context.Context) error {
+	// Find path to the current executable to determine how to update
+	ex, err := os.Executable()
+	if err != nil {
+		return err
+	}
+	term.Debugf(" - Executable: %s\n", ex)
+
+	ex, err = filepath.EvalSymlinks(ex)
+	if err != nil {
+		return err
+	}
+	term.Debugf(" - Evaluated: %s\n", ex)
+
+	if strings.Contains("brew/", ex) {
```

Comment:
I don't think this is sufficient. The homebrew prefix is `/usr/local` on intel machines:

> The script installs Homebrew to its default, supported, best prefix (/opt/homebrew for Apple Silicon, /usr/local for macOS Intel and /home/linuxbrew/.linuxbrew for Linux)

https://docs.brew.sh/Installation

I think I would create a little function to find the homebrew prefix (if homebrew is installed)

```go
func homebrewPrefix(ctx context.Context) (string) {
	output, err := exec.CommandContext(ctx, "brew", "config").Output()
	if err != nil {
		return ""
	}
	homebrewPrefix := ""
	// filter out the line which includes HOMEBREW_PREFIX
	for _, line := range strings.Split(string(output), "\n") {
		config_key := "HOMEBREW_PREFIX: "
		if strings.HasPrefix(line, config_key) {
			// remove the prefix from the line
			homebrewPrefix = strings.TrimPrefix(line, config_key)
			break
		}
	}
	return homebrewPrefix
}
```

Then do something like this:
```
	if strings.HasPrefix(ex, homebrewPrefix(ctx)) {
		term.Info(" * Updating defang via Homebrew...")
		return exec.CommandContext(ctx, "brew", "upgrade", "defang").Run()
	}
```